walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -25,9 +25,13 @@
 
     // Using linked list as removing the first element will take O(1) time
     LinkedList<Event> eventList = new LinkedList<Event>(events);
+
     // Collections.sort() -> O(nlogn)
     Collections.sort(eventList, Event.ORDER_EVENT_BY_START);
 
+    // Create a new list for traversal
+    LinkedList<Event> eventList2 = new LinkedList<Event>(eventList);
```

Comment:
Why is this second array necessary? I don't think you ever modify eventList, so you should be able to just re-use it.

Reply:
For some reason the eventList is being modified. When i do this
![image](https://user-images.githubusercontent.com/17647506/85329018-96fcf900-b486-11ea-9184-32a31056392e.png)
It returns an empty list

Reply:
Thoughts? I'm thinking maybe it's something in the list.removeFirst(), but since Java is pass by value shouldn't it not matter?

Reply:
After talking to support, they said that popping eventList actually modifies it, so making a copy is necessary. I would rather do this than convert my code into an iterator, if that's ok with you.

Reply:
Makes sense. But you should copy it into a local variable your `traverseThroughSchedule` function. In general, functions should not modify their parameters unless there's a good reason for it.